src/main/java/com/amazon/ion/impl/bin/IonRawBinaryWriter.java

```diff
@@ -380,15 +417,11 @@ interface ThrowingRunnable {
         this.preallocationMode = preallocationMode;
         this.isFloatBinary32Enabled = isFloatBinary32Enabled;
         this.buffer            = new WriteBuffer(allocator, this::endOfBlockSizeReached);
-        this.patchPoints       = new _Private_RecyclingQueue<>(512, PatchPoint::new);
-        this.containers        = new _Private_RecyclingStack<ContainerInfo>(
-            10,
-            new _Private_RecyclingStack.ElementFactory<ContainerInfo>() {
-                public ContainerInfo newElement() {
-                    return new ContainerInfo();
-                }
-            }
-        );
+        this.patchPoints       = new ArrayList<>(512);
+        this.patchPointsLength = 0;
+        this.containers        = new ArrayList<>(10);
```

Comment:
It's good when a new arbitrary constant is introduced to give some comment about its provenance, e.g. "chosen arbitrarily, subject to tuning" or "chosen by \<benchmarking process>/\<such and such analysis>"